wondrous-app/components/AddDocumentDialog/AddDocumentDialog.tsx

```diff
@@ -0,0 +1,170 @@
+
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import TextField from '@mui/material/TextField';
+
+import { CREATE_ORG_DOCUMENT, UPDATE_ORG_DOCUMENT } from 'graphql/mutations/documents';
+import { GET_ORG_DOCS } from 'graphql/queries/documents';
+import { URL_REGEX } from 'utils/constants';
+
+import DocModal from 'components/DocModal';
+import ImageUploader from 'components/ImageUploader';
+import DocPermissionSelect from 'components/DocPermissionSelect';
+
+import styles, { labelStyles, inputStyles } from './AddDocumentDialogStyles';
+
+const LabelTypography = styled(Typography)(labelStyles);
+const StyledTextField = styled(TextField)(inputStyles);
+
+const AddDocumentDialog = ({ open, onClose, title, orgId, category, document, pinned }) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+    setValue,
+  } = useForm();
+
+  const isEdit = !isEmpty(document);
+
+  const handleClose = () => {
+    reset();
+    onClose();
+  };
+
+  const [getOrgDocs] = useLazyQuery(GET_ORG_DOCS, {
+    variables: {
+      orgId,
+    },
+  });
+
+  const [createOrgDocument] = useMutation(CREATE_ORG_DOCUMENT, {
+    onCompleted: () => {
+      getOrgDocs();
+    },
+  });
+
+  const [updateDocument] = useMutation(UPDATE_ORG_DOCUMENT, {
+    onCompleted: () => {
+      getOrgDocs();
+    },
+  });
+
+  const onSubmitForm = (data) => {
+    if (isEdit) {
+      updateDocument({
+        variables: {
+          documentId: document.id,
+          input: {
+            description: data.description,
+            link: data.link,
+            title: data.title,
+            visibility: data.visibility,
+            previewPicture: data.previewPicture,
+          },
+        },
+      });
+
+      handleClose();
+      return;
+    }
+
+    createOrgDocument({
+      variables: {
+        input: {
+          ...data,
+          orgId: orgId,
+          categoryId: category.id,
+          ...(pinned && { pinned: pinned }),
+        },
+      },
+    });
+    handleClose();
+  };
+
+  useEffect(() => {
+    reset({ ...document });
+  }, [document]);
+
+  return (
+    <DocModal
+      open={open}
+      onClose={handleClose}
+      title="Add a Doc to"
+      highlightedText={title}
+      titleIcon={<Image src="/images/icons/addDoc.png" alt="folder icon" width={16} height={20} />}
+      onSubmit={handleSubmit(onSubmitForm)}
+      submitLabel={isEdit ? 'Edit Doc' : 'Add Doc'}
+    >
+      <form>
+        <Box mb={3.5}>
+          <LabelTypography>Link</LabelTypography>
+          <StyledTextField
+            name="link"
+            placeholder="Enter here a valid link"
+            {...register('link', {
+              required: 'link is required',
+              pattern: { value: URL_REGEX, message: 'Please enter a valid link' },
+            })}
+            fullWidth
+            helperText={errors.link && errors.link.message}
+            error={errors.link}
```

Comment:
Would we need to guard for the link field if errors are empty?

Reply:
Good point, added guards